rpcserver.go

```diff
@@ -1113,9 +1114,17 @@ func (r *rpcServer) GetInfo(ctx context.Context,
 		activeChains[i] = chain.String()
 	}
 
+	// Check if external IP addresses were provided to lnd and use them
+	// to set the URIs.
+	addrs := r.server.currentNodeAnn.Addresses
```

Comment:
I don't think this is thread safe, since the current node announcement can be changed by another thread in the server. Instead you could do 
```go
nodeAnn, err := r.server.genNodeAnnouncement(false)
...
```
do safely fetch a copy of the node announcement :)